Observer/NewsletterSubscribeObserver.php

```diff
@@ -45,16 +45,19 @@ public function execute(Observer $observer)
 
         /** @var Subscriber $subscriber */
         $subscriber = $observer->getDataObject();
-        if ($subscriber && ($subscriber->hasDataChanges() || $subscriber->isObjectNew())) {
+        if ($subscriber && $subscriber->isStatusChanged()) {
+            $subscriptionStatus = $subscriber->getStatus();
             $customer = $this->getCustomer($subscriber);
 
-            if ($subscriber->isSubscribed()) {
+            if ($subscriber->getId() && $subscriptionStatus === Subscriber::STATUS_SUBSCRIBED) {
```

Comment:
Just curious why we need to check the id?

Reply:
I was going off of the `isSubscribed` function from the core newsletter module: https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Newsletter/Model/Subscriber.php#L377. I removed that function call in favor of explicit status checks for subscribed and unsubscribed

Reply:
Awesome, sounds good!